aya-obj/src/obj.rs

```diff
@@ -711,10 +719,36 @@ impl Object {
         Ok(())
     }
 
-    fn parse_program(&self, section: &Section) -> Result<(Program, Function), ParseError> {
-        let prog_sec = ProgramSection::from_str(section.name)?;
-        let name = prog_sec.name().to_owned();
+    fn parse_programs(&mut self, section: &Section) -> Result<(), ParseError> {
+        let program_section = ProgramSection::from_str(section.name)?;
+        if let Some(syms) = self.symbols_by_section.get(&section.index) {
```

Comment:
Not sure what we should do here. For maps we return an error, but here we just
do nothing and skip the section. We should probably rename
NoSymbolsForMapsSection to MissingSectionSymbols and return it here too?

Reply:
In your last review you said it's valid for sections that to have no symbols...
Sections that contain BPF programs always have symbols though - which is why this patch works 🤔

For maps (and BTF maps) we know that both sections always have symbols, and we're able to derive the section type from the name - "maps" or ".maps".

Our detection logic for `ProgramSection` is basically a catch-all - i.e it wasn't one of the other enum variants and it's a  Text section with length > 0.

If we're 100% confident that will only match sections that contain BPF programs (and therefore have symbols) we can error out here too.

I'm not sure that's always the case though, so skipping the section seems sensible for now - we could always emit a warn log to help debug if that's easier.